### PR TITLE
Dilpreet | Test |UI Implementation of Activity logs

### PR DESCRIPTION
**Pull request by dilpreetsinghofficial**
https://app.clickup.com/t/2adthyp

### CODE REVIEW

File: apps/web-giddh/src/app/activity-logs/activity-logs.component.html [hunk 1]
```diff
@@ -70,54 +73,144 @@
...
+                                                [(ngModel)]="details.isChecked"
+                                                class="history-checkbox"
+                                                (change)="selectedItems($event, row2, details)"
+                                            >
+                                            </mat-checkbox>
+                                            <a
+                                                class="compare-json cursor-pointer"
+                                                *ngIf="details.isChecked"
+                                                (click)="compareJson(row2, details)"
+                                                >Compare</a
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/activity-logs/activity-logs.component.html]: Hard coded label, should be either from localeData or commonLocaleData

File: apps/web-giddh/src/app/activity-logs/activity-logs.component.ts [hunk 2]
```diff
@@ -298,6 +316,87 @@ export class ActivityLogsComponent implements OnInit, OnDestroy {
...
+     * @param {*} details
+     * @memberof ActivityLogsComponent
+     */
+    public selectedItems(event: MatCheckboxChange, row2: any, details: any): void {
+        if (!row2.selectedItems) {
+            row2.selectedItems = [];
+        }
+        if (event.checked) {
+            details.isChecked = true;
+            this.changeDetection.detectChanges();
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/activity-logs/activity-logs.component.ts]: Consider calling detectChanges at the end of the method as it is expensive operation

File: apps/web-giddh/src/assets/css/json-diff.scss [hunk 3]
```diff
@@ -0,0 +1,149 @@
+.jsondiffpatch-delta {
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/assets/css/json-diff.scss:1]: Consider using nesting of SCSS

File: package.json [hunk 4]
```diff
@@ -104,7 +104,10 @@
     "highcharts-angular": "^2.7.0",
     "is-cidr": "^4.0.0",
     "jquery": "3.5.1",
+    "json-diff": "^0.7.4",
```
Comment (the-prateek-mishra) [inline package.json]: Please check if all the new packages are being used

**Review by the-prateek-mishra — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/activity-logs/activity-logs.component.html [hunk 5]
```diff
@@ -70,54 +73,145 @@
...
+                    </td>
                 </ng-container>
-
-                <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
+                <ng-container matColumnDef="expandedDetail">
+                    <td
+                        mat-cell
+                        class="p-0 w-100 child-td"
+                        *matCellDef="let row2"
+                        [hidden]="!row2.isExpanded"
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/activity-logs/activity-logs.component.html]: rename row2 to rowHistory

File: apps/web-giddh/src/app/activity-logs/activity-logs.component.scss [hunk 6]
```diff
@@ -1,20 +1,63 @@
...
+        .activity-td {
+            width: 20%;
+        }
+        .history-btn {
+            display: none;
+        }
+        .activity-history {
+            min-height: 0;
+        }
+        tr:hover {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/activity-logs/activity-logs.component.scss]:
```suggestion
        tr {
          &:hover
```

File: apps/web-giddh/src/app/activity-logs/activity-logs.component.ts [hunk 7]
```diff
@@ -86,6 +89,14 @@ export class ActivityLogsComponent implements OnInit, OnDestroy {
     public universalDate: any;
     /** To show clear filter */
     public showDateReport: boolean = false;
+    /** To show clear filter */
+    public isDisabled: boolean = false;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/activity-logs/activity-logs.component.ts]: remove this var

File: apps/web-giddh/src/app/activity-logs/activity-logs.component.ts [hunk 8]
```diff
@@ -297,6 +311,83 @@ export class ActivityLogsComponent implements OnInit, OnDestroy {
...
+     *
+     * @param {*} entry Transaction object
+     * @memberof ActivityLogsComponent
+     */
+    public getHistory(event: any, row: any): void {
+        event.stopPropagation();
+        event.preventDefault();
+        row.isExpanded = !row.isExpanded;
+        if (!row.hasHistory) {
+            let activityObj = { entityId: row.entityId, entity: row.entity, count: 100 };
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/activity-logs/activity-logs.component.ts]:
```suggestion
            let activityObj = { entityId: row.entityId, entity: row.entity, count: 200 };
```

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Review by the-prateek-mishra — APPROVED** [review]

**Pull request merged**